thirdparty/DiskANN/src/pq_flash_index.cpp

```diff
@@ -293,20 +299,20 @@ namespace diskann {
 #ifdef EXEC_ENV_OLS
   template<typename T>
   void PQFlashIndex<T>::generate_cache_list_from_sample_queries(
-      MemoryMappedFiles &files, std::string sample_bin, _u64 l_search,
-      _u64 beamwidth, _u64 num_nodes_to_cache, uint32_t nthreads,
-      std::vector<uint32_t> &node_list) {
+      MemoryMappedFiles files, std::string sample_bin, _u64 l_search,
+      _u64 beamwidth, _u64 num_nodes_to_cache) {
 #else
   template<typename T>
   void PQFlashIndex<T>::generate_cache_list_from_sample_queries(
       std::string sample_bin, _u64 l_search, _u64 beamwidth,
-      _u64 num_nodes_to_cache, uint32_t nthreads,
-      std::vector<uint32_t> &node_list) {
+      _u64 num_nodes_to_cache) {
 #endif
     auto s = std::chrono::high_resolution_clock::now();
-    this->count_visited_nodes = true;
+    this->search_counter = 0;
     this->node_visit_counter.clear();
     this->node_visit_counter.resize(this->num_points);
+    reinterpret_cast<std::atomic<bool> &>(this->count_visited_nodes).exchange(true);
```

Comment:
here we don't use the value returned by `exchange`, and since both `exchange(read-modify-write)` and `store(write)` are atomic, there is no difference in terms of effect. 

Why is atomic read-modify-write more suitable?

Reply:
I think exchange is ： lock->read value to register -> write to mem -> unlock; store : lock -> wirte to mem -> unlock. The update of the counter is not strict, and store can be faster, so i update the code.